Extractor/Pushers/PusherUtils.cs

```diff
@@ -136,11 +136,13 @@ public static long GetTimestampValue(object value)
                         out _);
 
                 if (newMetaData.Any() && (old.Metadata == null
-                        || !newMetaData.All(kvp => old.Metadata.TryGetValue(kvp.Key, out var oldVal) && kvp.Value == oldVal)))
+                        || !newMetaData.All(kvp => old.Metadata.TryGetValue(kvp.Key, out var oldVal) && kvp.Value == oldVal))
+                    || config.Extraction.Deletes.Enabled && old.Metadata.ContainsKey(config.Extraction.Deletes.DeleteMarker))
```

Comment:
Might be a good idea to refactor this block into its own function or something else to make it less complex.